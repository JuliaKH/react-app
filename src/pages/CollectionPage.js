import React, {Component} from 'react';
import axios from "axios";
import './ImageList/ImageList.scss'

class CollectionPage extends Component {
    constructor(props) {
        super(props);
        this.state = {photos: []}
    }

    componentDidMount() {
        this.getChosenCollection();
    }

    componentWillReceiveProps(nextProps){
        this.getChosenCollection();
    }

    getChosenCollection = (id = this.props.match.params.id) => {
        const options = {
            params: {
                per_page: 20
            },
            headers: {
                Authorization: 'Client-ID 5110e0875d03049c42ef2483cf9a9ad53c6a0f46dd526e9ee18dca0c3c6a8f0b'
            }
        };
        axios
            .get(`https://api.unsplash.com/collections/${id}/photos`, options)
            .then(res => {
                this.setState({photos: res.data});
            })
            .catch(err => {
                console.log(err);
            });
    };
    render() {
        console.log(this.state);
        const photos = this.state.photos.map((photo) => {
            return <a href="#" key={photo.id}><img src={photo.urls.regular} alt={photo.alt_description} className="images__list-img"/></a>
        });
        return (
            <section className="images">
                <div className="container">
                    <div className="images__list">
                        {photos}
                    </div>
                </div>
            </section>
        );
    }
}

export default CollectionPage;
