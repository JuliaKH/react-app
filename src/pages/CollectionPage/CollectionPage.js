import React, {Component} from 'react';
import axios from "axios";
import '../ImageList/ImageList.scss'
import {Link} from "react-router-dom";
import ImageCard from "../ImageCard/ImageCard";

class CollectionPage extends Component {
    constructor(props) {
        super(props);
        this.state = {photos: []}
    }

    componentDidMount() {
        this.getChosenCollection(this.props.match.params.id)
    }

    componentWillReceiveProps(nextProps){
        this.getChosenCollection(nextProps.match.params.id);
    }

    getChosenCollection = (id) => {
        const options = {
            params: {
                per_page: 30
            },
            headers: {
                Authorization: 'Client-ID 5110e0875d03049c42ef2483cf9a9ad53c6a0f46dd526e9ee18dca0c3c6a8f0b'
            }
        };
        axios
            .get(`https://api.unsplash.com/collections/${id}/photos`, options)
            .then(res => {
                this.setState({photos: res.data});
            })
            .catch(err => {
            });
    };
    render() {
        const photos = this.state.photos.map((photo) => {
            return <Link to={`/images/${photo.id}`} key={photo.id}><img src={photo.urls.regular} alt={photo.alt_description} className="images__list-img"/></Link>
        });
        return (
            <section className="images">
                <div className="container">
                    <h2 className="images__title">{this.props.match.params.title}</h2>
                    <div className="images__list">
                        {photos}
                    </div>
                </div>
            </section>
        );
    }
}
CollectionPage.defaultProps = {
    match: { params: { code: 123 } }
};
export default CollectionPage;
