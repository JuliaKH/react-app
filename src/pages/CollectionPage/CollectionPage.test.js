import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CollectionPage from "./CollectionPage";
import { render } from 'enzyme'
import axios from "axios";

describe('CollectionPage ', () => {

    module.exports = {
        get: jest.fn(() => {
            return Promise.resolve({
                data: [
                    {
                        id: 1,
                        urls: {
                            regular: 'some.jpg'
                        },
                        alt_description: 'alt'
                    }
                ]
            })
        })
    };
    jest.mock('axios');


    it('should render correctly', () => {
        const wrapper = shallow(<CollectionPage/>);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should call componentDidMount', () => {
        jest.spyOn(CollectionPage.prototype, 'componentDidMount');
        const wrapper = render(<CollectionPage />);
        expect(CollectionPage.prototype.componentDidMount.mock.calls.length).toBe(1);
    });

    it('should fetch a image', () => {
        const getSpy = jest.spyOn(axios, 'get');
        const wrapper = shallow(<CollectionPage />);
        expect(getSpy).toBeCalled()
    });

});
