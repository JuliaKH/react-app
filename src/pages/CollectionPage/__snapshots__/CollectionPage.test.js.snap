// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage  should render correctly 1`] = `
<section
  className="images"
>
  <div
    className="container"
  >
    <h2
      className="images__title"
    />
    <div
      className="images__list"
    />
  </div>
</section>
`;

exports[`ImageList  should render correctly 1`] = `
<section
  className="images"
>
  <div
    className="container"
  >
    <h2
      className="images__title"
    />
    <div
      className="images__list"
    />
  </div>
</section>
`;
